pallets/capacity/src/lib.rs

```diff
@@ -477,6 +715,36 @@ impl<T: Config> Pallet<T> {
 		Ok(capacity)
 	}
 
+	fn increase_stake_and_issue_boost_capacity(
```

Comment:
I’ve noticed a significant overlap between `increase_stake_and_issue_capacity` and `increase_stake_and_issue_boost_capacity`. While I generally don’t mind code repetition if it enhances clarity, I’m a bit concerned in this case that we might accidentally make changes in one function and not the other, potentially leading to unintended bugs.

Reply:
additionally, is there a reason why the function `set_staking_account_and_lock` inside both `increase_stake_and_issue_capacity` and `increase_stake_and_issue_boost_capacity` is in a different order?

Reply:
I'll look at this again, but at one point it was making the code more complicated to combine the two, given what else has to happen when you boost versus staking MaximizedCapacity, however maybe there are some small optimizations.

Reply:
Thanks for taking another look! I get that combining the two functions might complicate things, especially with the differences between boosting and staking MaximizedCapacity. That said, having a single place for the shared logic could help prevent bugs from creeping in if the duplicated code gets updated differently. Happy to help just let me know.

Reply:
Pushed to an update. See #2207